tests/test_calculate.py

```diff
@@ -0,0 +1,206 @@
+        self.name = name
+        self.atoms = atoms
+        self.charge = charge
+        self.mult = mult
+
+
+class MockCalculation:
+    """Mock class for autode.calculation.Calculation"""
+
+    def __init__(self, name, molecule, method, keywords, n_cores):
+        self.name = name
+        self.molecule = molecule
+        self.method = method
+        self.keywords = keywords
+        self.n_cores = n_cores
+        self._energy_success = (
+            True  # Flag for controlling energy calculation success
+        )
+        self._gradient_success = (
+            True  # Flag for controlling gradient calculation success
+        )
+
+    def run(self):
+        """Simulate the run behavior."""
+        pass  # Assume the calculation runs successfully
+
+    def get_gradients(self):
+        """Return mock gradients or raise an error if _gradient_success is False"""
+        if self._gradient_success:
+            return MockUnitConversion(np.array([-0.1, 0.2, -0.3]))
+        else:
+            raise Exception('Could not get gradients')
+
+    def get_energy(self):
+        """Return mock energy or None if _energy_success is False"""
+        return MockUnitConversion(-1.0) if self._energy_success else None
+
+    def get_atomic_charges(self):
+        """Return mock partial charges"""
+        return np.array([0.5, -0.5])
+
+    @property
+    def output(self):
+        """Simulate output attribute with file lines"""
+
+        class Output:
+            file_lines = ['Some output log line'] * 50
+            exists = True
+
+        return Output()
+
+
+class MockUnitConversion:
+    """Mock class for autode's units conversion, simulating to()"""
+
+    def __init__(self, value):
+        self.value = value
+
+    def to(self, unit):
+        return self.value
+
+
+# Fixtures
+
+
+@pytest.fixture
+def configuration():
+    """Fixture for a mock Configuration object"""
+    config = Configuration(atoms=['H', 'O'], charge=0, mult=1)
+    config.forces = Forces()
+    config.energy = Energy()
+    config.partial_charges = None
+    return config
+
+
+@pytest.fixture
+def mock_method_and_keywords(monkeypatch):
+    """Fixture to mock _method_and_keywords function"""
+
+    def _mock_method_and_keywords(method_name):
+        return 'mock_method', 'mock_keywords'
+
+    monkeypatch.setattr(
+        'mlptrain.configurations.calculate._method_and_keywords',
+        _mock_method_and_keywords,
+    )
+
+
+@pytest.fixture
+def mock_autode(monkeypatch):
+    """Fixture to mock autode Species and Calculation"""
+    monkeypatch.setattr('autode.species.Species', MockSpecies)
+    monkeypatch.setattr('autode.calculation.Calculation', MockCalculation)
+
+
+@pytest.fixture
+def set_config(monkeypatch):
+    """Fixture to set required config values for ORCA and Gaussian keywords"""
+    Config.orca_keywords = ['PBE', 'def2-SVP', 'EnGrad']
+    Config.gaussian_keywords = ['B3LYP', '6-31G(d)', 'Force']
```

Comment:
ChatGPT, you forgot to unset the Config at the end of the test


```suggestion
    gauss_kws = Config.gaussian_keywords
    orca_kws = Config.orca_keywords 
    Config.orca_keywords = ['PBE', 'def2-SVP', 'EnGrad']
    Config.gaussian_keywords = ['B3LYP', '6-31G(d)', 'Force']
    yield
    Config.gaussian_keywords = gauss_kws
    Config.orca_keywords = orca_kws
```

https://docs.pytest.org/en/latest/how-to/fixtures.html#teardown-cleanup-aka-fixture-finalization